File: yurt-convention-api/src/main/java/com/github/liulus/yurt/convention/data/Pageable.java
package com.github.liulus.yurt.convention.data;

/**
 * @author liulu
 * @version V1.0
 * @since 2020/11/9
 */
public interface Pageable {

    /**
     * 页码
     */
    int getPageNum();

    /**
     * 每页显示记录数
     */
    int getPageSize();

    /**
     * 偏移量
     */
    int getOffset();

    /**
     * 是否关闭分页查询
     */
    boolean isDisablePage();

    /**
     * 是否查询count
     */
    default boolean isCount() {
        return true;
    }

}

File: yurt-convention-api/src/main/kotlin/com/github/liulus/yurt/convention/data/Pageable.kt
package com.github.liulus.yurt.convention.data

/**
 *
 * @author liulu
 * @version V1.0
 * @since 2020/10/29
 */
interface Pageable {

    /**
     * 页码
     */
    val pageNum: Int

    /**
     * 每页显示记录数
     */
    val pageSize: Int

    /**
     * 偏移量
     */
    val offset: Int

    /**
     * 是否查询count
     */
    val isCount: Boolean
        get() = true


}